spark-operator/src/main/java/org/apache/spark/k8s/operator/config/ConfigOption.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator.config;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import org.apache.spark.k8s.operator.utils.ModelUtils;
+
+/** Config options for Spark Operator. Supports primitive and serialized JSON */
+@RequiredArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode
+@ToString
+@Builder
+@Slf4j
+public class ConfigOption<T> {
```

Comment:
Hmm. So, is this duplication inevitable because we cannot take advantage of the existing and proven Apache Spark's `SparkConf` class due to the language difference (Scala/Java)? 

- https://github.com/apache/spark/blob/4a471cceebedd938f781eb385162d33058124092/core/src/main/scala/org/apache/spark/SparkConf.scala#L56

Reply:
yes, unfortunately the primitive conversion is quite painful here, hence a relatively lightweight conf abstraction is introduced 😞